packages/examples/sample_app/lib/main.dart

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_analytics_pinpoint/amplify_analytics_pinpoint.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3/amplify_storage_s3.dart';
+import 'package:flutter/material.dart';
+import 'package:sample_app/Pages/LoadingPage.dart';
+
+import 'Pages/LandingPage.dart';
+
+import 'amplifyconfiguration.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  Amplify amplify = new Amplify();
+  bool _isAmplifyConfigured = false;
+
+  @override
+  initState() {
+    super.initState();
+    _initAmplifyFlutter();
+  }
+
+  void _initAmplifyFlutter() async {
+    AmplifyAuthCognito auth = new AmplifyAuthCognito();
+    AmplifyStorageS3 storage = new AmplifyStorageS3();
+    AmplifyAnalyticsPinpoint analytics = new AmplifyAnalyticsPinpoint();
+
+    amplify.addPlugin(
+        authPlugins: [auth],
+        storagePlugins: [storage],
+        analyticsPlugins: [analytics]);
+
+    // Initialize AmplifyFlutter
+    await amplify.configure(amplifyconfig);
+    setState(() {
+      _isAmplifyConfigured = true;
+    });
+
+    AnalyticsEvent event = AnalyticsEvent("image_url_retrieved");
```

Comment:
The amplify-android/ios libs store signIn data so that the user's session persists upon app exit and restart.  We should demonstrate the ability to check the session at app startup, and enter a signedIn state depending on the result.

You can do something like this:

```
    var isSignedIn = false;
   ... 
    try {
      isSignedIn = await _isSignedIn();
    } on AuthError catch(e) {
      print("User is not signed in.");
    }
   ...
    Future<bool> _isSignedIn() async {
      final session = await Amplify.Auth.fetchAuthSession(
        options: CognitoSessionOptions(getAWSCredentials: false)
      );
      return session.isSignedIn;
    }
```

Reply:
You don't necessarily need to pass in the options parameter.  I left it in to educate the user.

Reply:
Added and tested your suggested code changes.